### DISCUSSION TITLE
Move to push towards 1.0.0

### DISCUSSION THREAD

**Discussion by filimon-danopoulos-stratsys** [modernweb-dev/web#2472] [General]
I propose a move towards 1.0.0 and seems like this is good forum discuss the pros and cons. I am not 100% familiar with the package structure but I assume this affects the entire monorepo. 

The main motivation would be to signify the maturity of the project. It seems feature complete and has been used in production for many years by many. Pushing towards 1.0.0 has a signal value in that this is a mature project. It also reflects a commitment to prevent breaking changes. The main drawback would be that more time has to be devoted towards backwards compatibility. 

As far as I am concerned the primary reason to remain in 0.X.Y is that it allows greater freedom in regards to breaking changes. On paper there is no real commitment to prevent breaking changes between minor releases without a proper 1.0.0 release. 

Assuming the push to 1.0.0 is reasonable it should probably be accompanied by some sort of breaking change policy to minimize the amount of labor that has to be put in as each subsequent breaking change is planned.

**Comment by trusktr** [comment]
@web/test-runner is excellent, I've been using it for a long time now (Starship UIs are tested with WTR, and I'm about to push an update for [Lume](https://lume.io/) that migrates from Karma to WTR very shortly and it has been incredible how much simplicity and edcellent dev experience and debuggability WTR brings with no clunkiness and hardly any issues especially on vanilla ES Modules), and people even avoid trying it out due to it being pre-1.0 (https://twitter.com/brophdawg11/status/1717275611188613158)

WTR is soooooooo incredibly better than Karma, that its like Karma v1000.

So at this point, I think a 1.0 version would really just mean "we promise not to change the API in a breaking way, this is what we're settling with for a while", but WTR has basically no stability problems.

I eliminated so so many problems by migrating from Karma to WTR, that's its a blessing I can't even describe. People just have to try it! And a 1.0 will entice them!
